dotnet/src/SemanticKernel.Abstractions/Functions/KernelArguments.cs

```diff
@@ -36,12 +37,32 @@ public KernelArguments()
     /// </summary>
     /// <param name="executionSettings">The prompt execution settings.</param>
     public KernelArguments(PromptExecutionSettings? executionSettings)
+        : this(executionSettings is null ? null : [executionSettings])
     {
-        this._arguments = new(StringComparer.OrdinalIgnoreCase);
+    }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KernelArguments"/> class with the specified AI execution settings.
+    /// </summary>
+    /// <param name="executionSettings">The prompt execution settings.</param>
+    public KernelArguments(IReadOnlyCollection<PromptExecutionSettings>? executionSettings)
+    {
+        this._arguments = new(StringComparer.OrdinalIgnoreCase);
         if (executionSettings is not null)
         {
-            this.ExecutionSettings = new Dictionary<string, PromptExecutionSettings>() { { PromptExecutionSettings.DefaultServiceId, executionSettings } };
+            var newExecutionSettings = new Dictionary<string, PromptExecutionSettings>(executionSettings.Count);
+            foreach (var settings in executionSettings)
+            {
+                var targetServiceId = settings.ServiceId ?? PromptExecutionSettings.DefaultServiceId;
+                if (newExecutionSettings.ContainsKey(targetServiceId))
+                {
+                    throw new ArgumentException("When adding multiple execution settings, the service id needs to be provided and be unique for each.");
```

Comment:
The service id doesn't need to be provided because when not the default one is used. If provided it must be unique and only one default is allowed. So either one long message or one for each case. The message should include the service id so the developer can see which one is causing the issue.